### PR TITLE
[Docs] Clarify caveats for phonetic filters replace option

### LINKED ISSUE #26921: Using phonetic analyzer affects queries with fuzziness
**Elasticsearch version** (`bin/elasticsearch --version`): 5.6.2 and 6.0.0-rc1 (maybe others)

**Plugins installed**: [analysis-phonetic]

**JVM version** (`java -version`): official docker containers from docker.elastic.co

**OS version** (`uname -a` if on a Unix-like system): Linux ubuntu-xenial 4.4.0-96-generic #119-Ubuntu SMP Tue Sep 12 14:59:54 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**: Match queries with fuzziness do not work as expected when a phonetic analyzer is used on the field.  Removing the analyzer from the example below restores the expected behavior of the fuzziness.

**Steps to reproduce**:

```python
from elasticsearch import Elasticsearch
import sys

es = Elasticsearch(hosts=[sys.argv[1]])

index_body = {
    'settings': {
        'number_of_shards': 1
    },
    'mappings': {
        'test': {
            'properties': {
                'last_name': {
                    'type': 'text',
                    'analyzer': 'phonetic_analyzer',
                },
            },
        },
    },
    'settings': {
        'index': {
            'analysis': {
                'analyzer': {
                    'phonetic_analyzer': {
                        'filter': ['lowercase', 'phonetic_filter'],
                        'tokenizer': 'standard',
                        'type': 'custom'
                    },
                },
                'filter': {
                    'phonetic_filter': {
                        'replace': 'false',
                        'type': 'phonetic',
                    },
                },
            },
        },
    },
}
create_result = es.indices.create(index='test', body=index_body)

index_result = es.index(index='test', doc_type='test',
        body={'last_name': 'CONNE'},
        refresh=True)

index_result = es.index(index='test', doc_type='test',
    body={'last_name': 'CONNER'},
    refresh=True)

search_body = {
    'query': {
        'match': {
            'last_name': {
                'query': 'CONNER',
                'fuzziness': 'auto',
            },
        },
    },
    'explain': True,
}
search_result = es.search(index='test', doc_type='test',
        body=search_body)

print("This should have returned two hits, but only returned %s." %
        search_result['hits']['total'])  # 1

print("We only matched record with name: %s" %
        search_result['hits']['hits'][0]['_source']['last_name'])  # CONNER
print("This should have also matched the CONNE record too!")
```

### PR DESCRIPTION
**Pull request by cbuescher**
The `replace` option in the phonetic token filter can have suprising side
effects, e.g. such as described in #26921. This PR adds a note to be mindful
about such scenarios and offers alternatives to using the `replace` option.

Closes #26921

### CODE REVIEW

**Review by jtibshirani — COMMENTED** [review]
Thanks @cbuescher! I made note of a typo and also had a small suggestion: it might be helpful to add one line to the last paragraph of the [match query docs about fuzziness](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query.html#query-dsl-match-query-fuzziness), noting that the issue applies not only to synonyms but to any token filter that produces multiple tokens at the same position.

**Comment by cbuescher** [comment]
@jtibshirani thanks for the review, I corrected the typo, added some links and reprhased the paragraph a bit. 

I don't completely get your comment about adding to the last paragraph of the match query docs about fuzziness. The issue this documentation is trying to adress is to be aware of the fact that stacked token like the synonym filter produces can have unwanted side-effects, in this case most notably that match queries skip applying fuzziness to synonym queries the the original query is rewritten to when we have stacked token in the same position. The paragraph you linked talks about generating phrases which sounds related, but in the case of the example that is given in https://github.com/elastic/elasticsearch/issues/26921#issuecomment-335087296 the `auto_generate_synonyms_phrase_query` doesn't really make a difference in the output. I have problems relating the two things, so did you have any specific in minf to add to the  "match query docs about fuzziness" paragraph?

File: docs/plugins/analysis-phonetic.asciidoc [hunk 1]
```diff
@@ -65,6 +65,14 @@ GET phonetic_sample/_analyze
 
 <1> Returns: `J`, `joe`, `BLKS`, `bloggs`
 
+It is important to note that `"replace": false` can lead to unexpected behavior since
+the original and the phonetically analyzed version are both kept at the same token position.
+Some queries handle these stacked tokens in special ways. For example, the fuzzy `match`
+query does not apply <<fuzziness>> to stacked synonym tokens. This can lead to issues that are
```
Comment (jtibshirani) [inline docs/plugins/analysis-phonetic.asciidoc]: I tried to build these plugin docs, but I get the error " IDREF attribute linkend references an unknown ID "fuzziness"".

**Review by jtibshirani — COMMENTED** [review]
Sorry for the confusion, I meant the last paragraph in the section on fuzziness (not the last paragraph of the document overall). The paragraph ends with "Note that fuzzy matching is not applied to terms with synonyms, as under the hood these terms are expanded to a special synonym query that blends term frequencies, which does not support fuzzy expansion.".

This is just a suggestion, happy to approve + merge after the build issue is fixed.

**Comment by cbuescher** [comment]
>  I meant the last paragraph in the section on fuzziness

Thanks for the clarification, that makes a lot of sense to me now. Will look into it tomorrow, also into the plugin-docs issue you see (although the tests seem to pass, so I guess we're good)

**Comment by jtibshirani** [comment]
👍 will take another look tomorrow. I'm not sure our docs check actually runs all docs builds. For example, in the past I accidentally introduced an issue into the HLRC docs although the PR passed the docs checks.

File: docs/plugins/analysis-phonetic.asciidoc (hunk 1 above)
Comment (cbuescher) [reply docs/plugins/analysis-phonetic.asciidoc]: Yep, you are right, doesn't work here as well. I changed it to using "{ref}" which is what I think we use in cases where we link between two "books" in the docs.

**Comment by cbuescher** [comment]
@jtibshirani added the note and corrected the link, this should be good to go now once the tests pass.

**Review by jtibshirani — APPROVED** [review]

**Pull request merged**